numpy/core/getlimits.py

```diff
@@ -482,9 +482,12 @@ class finfo:
     _finfo_cache = {}
 
     def __new__(cls, dtype):
-        obj = cls._finfo_cache.get(dtype)  # most common path
-        if obj is not None:
-            return obj
+        try:
+            obj = cls._finfo_cache.get(dtype)  # most common path
+            if obj is not None:
+                return obj
+        except TypeError:
+            obj = None
```

Comment:
```suggestion
            pass
```
Anyway, this looks fine to me and should fix the regression fine.  Thanks for the test.
(this was just a silly small nit)